### PR TITLE
Transfer Client.CLI and Client.Core to use a Logger

### PR DESCRIPTION
**Pull request by KanstantsinTaukach**

### CODE REVIEW

File: Client/Client.Core/Client.Core.Static/Client.hpp [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #include "Network/Message.hpp"
 #include "Network/SafeQueue.hpp"
 
+#include <FileLogger.hpp>
```
Comment (Orange-hanter) [inline Client/Client.Core/Client.Core.Static/Client.hpp:7]: It's better to move FileLogger header to cpp part. It should improve build time and avoid unnecessary dependencies.

File: Client/Client.Core/Client.Core.Static/CMakeLists.txt [hunk 2]
```diff
@@ -12,18 +12,12 @@ endif ()
 
 add_library(${TARGET} STATIC ${SOURCE_FILES})
 
-target_include_directories(${TARGET}
```
Comment (Orange-hanter) [inline Client/Client.Core/Client.Core.Static/CMakeLists.txt]: Please, do not update format of CMake unless you have a reason for this

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

File: Client/Client.CLI/App.cpp [hunk 3]
```diff
@@ -9,80 +9,102 @@ App::~App()
...
 }
 
 void App::onMessageHistoryAnswer(const std::vector<Network::MessageInfo>& messages)
 {
     for (const auto& item : messages)
     {
-        std::cout << item.message << '\n';
+        std::ostringstream out;
+        out << item.message;
+        Base::Logger::FileLogger::getInstance().log(out.str(), Base::Logger::LogLevel::INFO);
```
Comment (fisyamaloy) [inline Client/Client.CLI/App.cpp:88]: Is it necessary to log each message? Maybe it is better to log only row such as "Message history recieved"

**Comment by KliKonNik** [comment]
Probably it's better to use **std::string()** instead of **out.str()**, isn't it?

File: Client/Client.Qt/ConnectionManager.cpp [hunk 4]
```diff
@@ -73,11 +81,11 @@ void ConnectionManager::onReplyStoreAnswer(Utility::StoringReplyCodes storingRep
 {
     if (storingReplyCode == Utility::StoringReplyCodes::SUCCESS)
     {
-        std::cout << "SUCCESS sending" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("SUCCESS sending", Base::Logger::LogLevel::INFO);
     }
     else if (storingReplyCode == Utility::StoringReplyCodes::FAILED)
     {
-        std::cout << "FAILED sending" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("FAILED sending", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.Qt/ConnectionManager.cpp]: Wrong Log lever. In this case should be ERROR

File: Client/Client.Qt/ConnectionManager.cpp [hunk 5]
```diff
@@ -87,11 +95,11 @@ void ConnectionManager::onMessageReactionAnswer(const Utility::ReactionMessageCo
 {
     if (reactionCode == Utility::ReactionMessageCodes::SUCCESS)
     {
-        std::cout << "Reaction update SUCCESS\n";
+        Base::Logger::FileLogger::getInstance().log("Reaction update SUCCESS", Base::Logger::LogLevel::INFO);
     }
     else if (reactionCode == Utility::ReactionMessageCodes::FAILED)
     {
-        std::cout << "Reaction update FAILED\n";
+        Base::Logger::FileLogger::getInstance().log("Reaction update FAILED", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.Qt/ConnectionManager.cpp]: same

File: Client/Client.Qt/ConnectionManager.cpp [hunk 6]
```diff
@@ -120,15 +128,15 @@ void ConnectionManager::onUserMessageDeleteAnswer(const Utility::DeletingMessage
 {
     if (deletingCode == Utility::DeletingMessageCodes::SUCCESS)
     {
-        std::cout << "SUCCESS DELETING" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("SUCCESS DELETING", Base::Logger::LogLevel::INFO);
     }
     else if (deletingCode == Utility::DeletingMessageCodes::FAILED)
     {
-        std::cout << "FAILED DELETING" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("FAILED DELETING", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.Qt/ConnectionManager.cpp]: same

File: Client/Client.Qt/ConnectionManager.cpp [hunk 7]
```diff
@@ -120,15 +128,15 @@ void ConnectionManager::onUserMessageDeleteAnswer(const Utility::DeletingMessage
...
     }
     else
     {
-        std::cout << "UNKNOWN deleting message code" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("UNKNOWN deleting message code", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.Qt/ConnectionManager.cpp]: Loglevel::WARNING

File: Client/Client.Qt/Pages/RegistrationPage.cpp [hunk 8]
```diff
@@ -50,7 +50,7 @@ RegistrationPage::RegistrationPage(QWidget* parent) : Page(parent)
 
         if (email.empty() || login.empty() || password.empty() || repeatPassword.empty())
         {
-            std::cout << "some field is empty" << std::endl;
+            Base::Logger::FileLogger::getInstance().log("some field is empty", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.Qt/Pages/RegistrationPage.cpp]: Here should be WARNING

File: Client/Client.Core/Client.Core.Static/Client.cpp [hunk 9]
```diff
@@ -417,121 +422,131 @@ void Client::loop()
             break;
 
             default:
-                std::cerr << "[Client][Warning] unimplemented[" << uint32_t(message.mHeader.mMessageType) << "]\n";
+                Base::Logger::FileLogger::getInstance().log
+                (
+                    "[Client][Warning] unimplemented[" + std::to_string(uint32_t(message.mHeader.mMessageType)) + "]",
```
Comment (Orange-hanter) [inline Client/Client.Core/Client.Core.Static/Client.cpp]: **[Client][Warning]** are superfluous. Also, the type of log level should be WARNING

File: Client/Client.Core/Client.Core.Static/Client.cpp [hunk 10]
```diff
@@ -417,121 +422,131 @@ void Client::loop()
...
+                );
         }
     }
 }
 
 void Client::onLoginAnswer(bool success)
 {
     (void)(success);
-    std::cerr << "[Client][Warning] login answer is not implemented\n";
+    Base::Logger::FileLogger::getInstance().log("[Client][Warning] login answer is not implemented", Base::Logger::LogLevel::ERR);
```
Comment (Orange-hanter) [inline Client/Client.Core/Client.Core.Static/Client.cpp]: The same as above

File: Client/Client.Core/Client.Core.Static/Client.cpp [hunk 11]
```diff
@@ -417,121 +422,131 @@ void Client::loop()
...
 }
 
-void Client::onServerAccepted() { std::cerr << "[Client][Warning] server accepted is not implemented\n"; }
+void Client::onServerAccepted()
+{
+    Base::Logger::FileLogger::getInstance().log("[Client][Warning] server accepted is not implemented", Base::Logger::LogLevel::ERR);
```
Comment (Orange-hanter) [inline Client/Client.Core/Client.Core.Static/Client.cpp]: same

File: Client/Client.CLI/MainClient.cpp [hunk 12]
```diff
@@ -106,15 +106,16 @@ int main(int argc, char** argv)
 
                 if (email.empty() || login.empty() || password.empty() || repeatPassword.empty())
                 {
-                    std::cout << "some field is empty" << std::endl;
+                    Base::Logger::FileLogger::getInstance().log("some field is empty", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.CLI/MainClient.cpp]: Warning Log level

File: Client/Client.CLI/App.cpp [hunk 13]
```diff
@@ -9,80 +9,102 @@ App::~App()
...
 {
     if (storingMessageCode == Utility::StoringMessageCodes::SUCCESS)
     {
-        std::cout << "SUCCESS sending" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("SUCCESS sending", Base::Logger::LogLevel::INFO);
     }
     else if (storingMessageCode == Utility::StoringMessageCodes::FAILED)
     {
-        std::cout << "FAILED sending" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("FAILED sending", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.CLI/App.cpp]: Here should be ERROR

File: Client/Client.CLI/App.cpp [hunk 14]
```diff
@@ -9,80 +9,102 @@ App::~App()
...
     }
     else
     {
-        std::cout << "Unknown StoringMessageCode" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("Unknown StoringMessageCode", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.CLI/App.cpp]: And here WARNING

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

File: Client/Client.CLI/App.cpp [hunk 15]
```diff
@@ -9,80 +9,102 @@ App::~App()
...
     }
     else if (registrationCode == Utility::RegistrationCodes::EMAIL_ALREADY_EXISTS)
     {
-        std::cout << "User with such email already exists" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("User with such email already exists", Base::Logger::LogLevel::INFO);
     }
     else
     {
-        std::cout << "Unknown RegistrationCode" << std::endl;
+        Base::Logger::FileLogger::getInstance().log("Unknown RegistrationCode", Base::Logger::LogLevel::INFO);
```
Comment (Orange-hanter) [inline Client/Client.CLI/App.cpp]: warning level

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

**Review by Orange-hanter — COMMENTED** [review]
Client CLI need more attention

**Review by Orange-hanter — APPROVED** [review]

**Comment by Orange-hanter** [comment]
@memorisecodead @ivanboika please review this!

**Review by ivanboika — APPROVED** [review]

**Pull request merged**